dev-docs/roadmaps/tile-layer-roadmap.md

```diff
@@ -0,0 +1,119 @@
+
+However light usage of the layer quickly discovers that a number of things are lacking.
+- key features like caching, request scheduling and additive parent-child refinements have now been added to the `Tile3DLayer` but are missing in the similar but simpler 2D `TileLayer`.
+- A number of ambitious user PRs proposing to extend the `TileLayer` in new directions show how important this layer is to deck.gl users. Without a clear roadmap it is hard to provide guidance on and accept these PRs in a timely way.
+- ...
+
+
+## Goals and Aspirations
+
+To be able to provide fair and consistent feedback on the ambitious PRs and feature requests being opened on the `TileLayer`, it is important to define what the layer is supposed to be and not to be.
+
+Goals
+- High performance tile loading
+- Flexibility to support various tile format
+- ...
+
+Non-goals
+- The `TileLayer` is NOT indended to be a full replacement for a base map (e.g. a mapbox base map) as that involves a long tail of features to achieve the required level of polish).
+
+## Current Issues
+
+### Known Issues with v8.0 Tile Layer
+
+- BUG: Parent tiles are not hidden - When opacity != 1 overlapping child/parent tiles are visible
+- Tile caching scheme is naive
+- No request scheduling - tiles no longer in view (e.g. after quick pans) can still be loaded, request queue floods with "low priority" tile requests starving other loading, ...
+- No visual optimizations (e.g. load tiles from center out, blend in child-tiles, ...)
+
+
+### PRs
+
+At the time of writing, the following PRs were open. This roadmap tries to understand what these PRs want to achieve, then proposes a number of improvements to the `TileLayer`, then contains recommendations for these PRs in light of the proposals:
+
+- [`MVTTileLayer`](https://github.com/uber/deck.gl/pull/3935) - Highlights: Mapbox vector tile support - Worker loading of mapbox vector tiles - Better/Customized Tile Cache - Composite Geometry...
+- [Non-geospatial `TileLayer`](https://github.com/uber/deck.gl/pull/4117) - Provide the necessary hooks in to control coordinate mapping to tiles  `tileSize`, `tile2boundingBox`, `getTileIndices`,
+- [TerrainLayer](https://github.com/uber/deck.gl/pull/3984) - An example of `TileLayer` used to display terrain. Something we want to make easy (at least easier) for users.
+
+## Proposals
+
+### Proposal: Breakout out `TileLayer` tile handling into `Tileset2D` class.
+
+- Would follow conventions/established pattern in `Tileset3D` in loaders.gl, making it easier for a maintainer to work on both layers.
+- Move `Tileset2D` to loaders.gl (easier maintenance)?
+- Use same API across `Tileset2D` and `Tileset3D`, e.g. for setting viewStates
+- Users who need to configure tile loading would not need to subclass the entire layer, just the `Tileset2D` class.
+
+### Proposal: data prop set to `Tileset` class
```

Comment:
I think we have settled on offering another prop to override `Tileset2D` with a custom class.

Reply:
Yes. This document is dated Jan 2020, I think it is OK to land it as-is, as a historical doc. I may refresh it part of ongoing work and will update this then.